pkg/controller/jobframework/reconciler.go

```diff
@@ -582,20 +582,34 @@ func (r *JobReconciler) IsParentJobManaged(ctx context.Context, jobObj client.Ob
 	if err := r.client.Get(ctx, client.ObjectKey{Name: owner.Name, Namespace: namespace}, parentJob); err != nil {
 		return false, errors.Join(ErrWorkloadOwnerNotFound, err)
 	}
-	return QueueNameForObject(parentJob) != "", nil
+	if QueueNameForObject(parentJob) != "" {
+		return true, nil
+	}
+	return r.IsAncestorJobManaged(ctx, parentJob, namespace)
```

Comment:
I see the recursive invocation in `IsAncestorJobManaged` and `getAncestorWorkload`. Could we instead have one function `getAncestoryJob`, which is called both by `IsAncestorJobManaged` and `getAncestorWorkload`?

Reply:
I tried a couple ways of doing this, but one is recursively looking for the queue-name label and the other is recursively looking for the workload.  In "real" scenarios these are pretty equivalent, but they aren't the same for unit tests.  So trying to merge the code either results in needing to throw out/redo a non-trivial number of tests or in writing a helper function that contains too much complexity because it has to do one or the other depending on who invokes it.  The duplicated code looks to be the cleanest option.